src/config.rs

```diff
@@ -56,6 +57,7 @@ impl Config {
 
             prefix: prefix.clone(),
             registry_index_path: env("REGISTRY_INDEX_PATH", prefix.join("crates.io-index"))?,
+            registry_url: maybe_env("REGISTRY_URL")?,
```

Comment:
Having both `--crate-registry` and `REGISTRY_URL` seems confusing. Could you change this to only one or the other?